### DISCUSSION TITLE
How turn all external hyperlinks into footnotes in PDF format?

### DISCUSSION THREAD

**Discussion by nichtich** [quarto-dev/quarto-cli#10355] [Q&A] [answered]
### Description

We create a book with quarto with online version in HTML and a printed version via PDF. The default (blue) display of hyperlinks in PDF is not suitable for a printed book. External links should be shown as footnotes instead, unless the URL is displayed in text.

````qmd
---
title: Example
format: pdf
number-sections: true
---

## Demo {#sec-demo}

This is an example with a [link to Wikipedia](https://en.wikipedia.org/). The URL should be should in a footnote.

A plain link such as <https://quarto.org> should not be shown as footnote, neither internal links such as to @sec-demo.
````

**Comment by mcanouil** [comment]
Use the markdown footnote syntax, see <https://quarto.org/docs/authoring/markdown-basics.html#footnotes>.
I am not sure to follow why you write link syntax when you want footnote.

Or use a LUA filter that read the document as AST and change the elements in whatever way you need.

**Reply by nichtich** [reply]
Thanks! Links should be shown inline in HTML and as footnote in PDF. Seems like a Lua filter is the way to go indeed:

~~~lua
local utils = require 'pandoc.utils'

function Link(link)
  if FORMAT == "latex" and utils.stringify(link.content) ~= link.target then
    return { link.content, pandoc.Note(link.target) }
  end
end
~~~

Unfortunately this gives error message:

> Inline, list of Inlines, or string expected, got table

When changed one line to this:

```lua
return link.content, pandoc.Note(link.target)
```

then no footnote is added at all (?)

**Reply by cderv** [reply]
> I am not sure to follow why you write link syntax when you want footnote.

I believe this is special case between what you want (type of content) and how you want it to be shown. Especially across formats between printed or non-printed content. 

So you want a link, so you want link syntax. In HTML, you can have clickable on screen. In PDF, you won't have the link information because it is printed. It would be the same in printing HTML to PDF. 

This is why usually you can have options to change the form of your content like this. In pandoc, `links-as-notes` does this (see [below answer](https://github.com/quarto-dev/quarto-cli/discussions/10355#discussioncomment-10124834)) . We support this feature. Quarto even allows more position with adding some value to `reference-location` like margin. 

````markdown
---
title: Example
format: 
  pdf:
    links-as-notes: true
    reference-location: margin
number-sections: true

---

## Demo {#sec-demo}

This is an example with a [link to Wikipedia](https://en.wikipedia.org/). The URL should be should in a footnote.

A plain link such as <https://quarto.org> should not be shown as footnote, neither internal links such as to @sec-demo.

````

![image](https://github.com/user-attachments/assets/ec1b1fe0-0855-43d4-b5b0-e9db64554a35)

Having option like this to separate the content type, and how it is rendered is definitely something in the scope of Quarto. 

Hope this helps understand why these type of use case can happen.

**Comment by cderv** [comment, marked as answer]
The option `links-as-notes` should be what you are looking for: https://quarto.org/docs/reference/formats/pdf.html#footnotes
It is initially a [Pandoc feature](https://pandoc.org/MANUAL.html#links), and it should still work in Quarto. 

Did you try this already ? 

````markdown
---
title: Example
format: 
  pdf:
    links-as-notes: true
number-sections: true

---

## Demo {#sec-demo}

This is an example with a [link to Wikipedia](https://en.wikipedia.org/). The URL should be should in a footnote.

A plain link such as <https://quarto.org> should not be shown as footnote, neither internal links such as to @sec-demo.

````
![image](https://github.com/user-attachments/assets/3428e47b-195e-40bf-801e-c61a47855158)

**Reply by mcanouil** [reply]
Pandoc had the option 🤯 (there are too many hidden gems in Pandoc^^)
